transformers/events/median_drop/transformer_test.go

```diff
@@ -0,0 +1,95 @@
+package median_drop_test
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/lib/pq"
+	"github.com/makerdao/vdb-mcd-transformers/test_config"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/events/median_drop"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/shared"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/shared/constants"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/test_data"
+	"github.com/makerdao/vulcanizedb/libraries/shared/factories/event"
+	"github.com/makerdao/vulcanizedb/pkg/core"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Median drop transformer", func() {
+	var (
+		transformer = median_drop.Transformer{}
+		db          = test_config.NewTestDB(test_config.NewTestNode())
+	)
+
+	BeforeEach(func() {
+		test_config.CleanTestDB(db)
+	})
+
+	It("returns err if log is missing topics", func() {
+		incompleteLog := core.EventLog{}
+		_, err := transformer.ToModels(constants.MedianABI(), []core.EventLog{incompleteLog}, db)
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("returns err if log is missing data", func() {
+		badLog := core.EventLog{
+			Log: types.Log{
+				Topics: []common.Hash{{}, {}, {}, {}},
+			}}
+
+		_, err := transformer.ToModels(constants.MedianABI(), []core.EventLog{badLog}, db)
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("convert a log with 4 accounts to an insertion model", func() {
+		models, err := transformer.ToModels(constants.MedianABI(), []core.EventLog{test_data.MedianDropLogWithFiveAccounts}, db)
+		Expect(err).NotTo(HaveOccurred())
+
+		a1Bytes, aErr := shared.GetLogNoteArgumentAtIndex(2, test_data.MedianDropLogWithFiveAccounts.Log.Data)
+		Expect(aErr).NotTo(HaveOccurred())
+		address0 := common.BytesToAddress(a1Bytes).Hex()
+
+		a2Bytes, a2Err := shared.GetLogNoteArgumentAtIndex(3, test_data.MedianDropLogWithFiveAccounts.Log.Data)
+		Expect(a2Err).NotTo(HaveOccurred())
+		address1 := common.BytesToAddress(a2Bytes).Hex()
+
+		a3Bytes, a3Err := shared.GetLogNoteArgumentAtIndex(4, test_data.MedianDropLogWithFiveAccounts.Log.Data)
+		Expect(a3Err).NotTo(HaveOccurred())
+		address2 := common.BytesToAddress(a3Bytes).Hex()
+
+		a4Bytes, a4Err := shared.GetLogNoteArgumentAtIndex(5, test_data.MedianDropLogWithFiveAccounts.Log.Data)
+		Expect(a4Err).NotTo(HaveOccurred())
+		address3 := common.BytesToAddress(a4Bytes).Hex()
+
+		expectedModel := test_data.MedianDropModelWithFiveAccounts()
```

Comment:
Is this model named correctly? Seems like 4 rather than 5 accounts, right?

Reply:
I believe the idea with this test is to demonstrate that if `drop` was passed an array of 5+ addresses, then the event truncates the array partway through the 5th address-- looking at the log's `Data`, this appears to be an example of that kind of event log. Might be worth updating the test's docstring to clarify what's happening.

Reply:
Ah makes sense - I guess we probably don't need that data (and could potentially grab it off the log if we wanted it in the future?). Cool that the log contains array length + item length - I imagine that could prove useful for general purpose helpers built around parsing arrays in events

Reply:
word yeah sounds like a docstring update is in order - was just confused if we were dealing with four or five entries in the array

Reply:
:point_up: exactly this. I forgot to update the docstring